app/controllers/admin/courses_controller.rb

```diff
@@ -0,0 +1,56 @@
+class Admin::CoursesController < ApplicationController
+  layout 'layouts/admin'
+  before_action :get_course, only:[:edit, :update, :destroy]
+
+  def index
+    @courses = Course.paginate(page: params[:page], per_page: 2)
+  end
+
+  def new
+    @course = Course.new
+  end
+
+  def create
+    @course = Course.new(course_params)
+    if @course.save
+      flash[:success] = "Add success"
+      redirect_to admin_courses_path
+    else
+      redirect_to  new_admin_course_path
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @course.update course_params
+      flash[:success] = "Profile updated"
+      redirect_to admin_courses_path
+    else
+      flash[:danger] = "Profile update fail"
+      render edit_admin_course_path
+    end
+  end
+
+  def destroy
+    if @course.destroy
+      flash[:success] = "Delete success"
+      redirect_to admin_courses_path
+    else
+      flash[:danger] = "Delete fail"
+      redirect_to admin_courses_path
+  end
+
+  private
+    def course_params
+      params.require(:course).permit(:name, :description)
+    end
+
+    def get_course
+      @course = Course.find_by_id(params[:id])
+      return if @course
+      flash[:danger] = "User not found"
+      redirect_to admin_courses_path
```

Comment:
nên làm 1 page common not found 
khi k tìm thấy thì render ra page đó 
sau mỗi dòng return, next thì add 1 line trống